test/test_dist_autograd.py

```diff
@@ -0,0 +1,20 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import unittest
+import torch.distributed.autograd as dist_autograd
+
+class TestDistAutograd(unittest.TestCase):
+
+    def test_autograd_context(self):
+        worker_id = 16
+        dist_autograd.init(worker_id)
+        with dist_autograd.context() as context_id:
+            self.assertEqual(context_id, dist_autograd.retrieve_context(context_id).context_id())
+            # First 16 bits should be worker_id.
+            self.assertEqual(worker_id, context_id >> 48)
```

Comment:
Is this actually documented behavior that we are committing to permanently, or do we actually just want the ID to be an opaque integer that we have full control over?